dataikuapi/dss/ml.py

```diff
@@ -643,6 +652,351 @@ def __repr__(self):
                                                           msg=self._internal_dict["message"])
 
 
+class DSSMLAssertionsParams(object):
+    """
+    Object that represents parameters for all assertions of a ml task
+    Do not create this object directly, use :meth:`DSSPredictionMLTaskSettings.get_assertions_params()` instead
+    """
+
+    def __init__(self, data):
+        self._internal_dict = data
+
+    def get_raw(self):
+        """
+        Gets the raw dictionary of the assertions parameters
+        :rtype: dict
+        """
+        return self._internal_dict
+
+    def get_assertion(self, assertion_name):
+        """
+        Gets a :class:`DSSMLAssertionParams` representing the parameters of the assertion with the
+        provided name (or None if no assertion has that name)
+        :param str assertion_name: Name of the assertion
+        :rtype: :class:`DSSMLAssertionParams` or None if no assertion has that name
+        """
+        for assertion_dict in self._internal_dict["assertions"]:
+            if assertion_dict["name"] == assertion_name:
+                return DSSMLAssertionParams(assertion_dict)
+        return None
+
+    def add_assertion(self, assertion_params):
+        """
+        Adds parameters of an assertion to the assertions parameters of the ml task.
+        :param object  assertion_params: A :class:`DSSMLAssertionParams` representing parameters of the assertion
+        """
+        if not isinstance(assertion_params, DSSMLAssertionParams):
+            raise ValueError('Wrong type for assertion parameters: {}'.format(type(assertion_params)))
+
+        self._internal_dict["assertions"].append(assertion_params._internal_dict)
+
+    def delete_assertion(self, assertion_name):
+        """
+        Deletes the assertion parameters of the assertion with the provided name from the assertions parameters of the ml task.
+        Raises a ValueError if no assertion with the provided name was found
+        :param str assertion_name: Name of the assertion
+        """
+        for idx, assertion_dict in enumerate(self._internal_dict["assertions"]):
+            if assertion_dict["name"] == assertion_name:
+                del self._internal_dict["assertions"][idx]
+                return
+        raise ValueError('No assertion found with name: {}'.format(assertion_name))
+
+
+class DSSMLAssertionParams(object):
+    """
+    Object that represents parameters for one assertion
+    Do not create this object directly, use :meth:`DSSMLAssertionsParams.get_assertion(assertion_name)` or
+    `from_parts(name, a_filter, condition)` instead
+    """
+    def __init__(self, data):
+        self._internal_dict = data
+
+    def __repr__(self):
+        return u"DSSMLAssertionParams(\n    name= '{name}',\n    condition= {condition},\n    filter= {filter}\n)".format(
+            name=self.name, filter=pprint.pformat(self.filter).replace('\n', '\n            '), condition=self.condition)
```

Comment:
Not certain we want multiline repr, as it makes it harder to grep in logs. IMO acceptable to have `DSSMLAssertionParams(<internal dict repr>)`